### PR TITLE
[SPARK-41507][SQL] Correct group of collection_funcs and fix description of `ExpressionDescription#group`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr change the function group in `collectionOperations.scala`:

-  ArraySize: `collection_funcs` -> `array_funcs` due to it only used for `Array`
- ElementAt: `array_funcs` -> `collection_funcs` due to it used for both `Array` and `Map`
- TryElementAt: `array_funcs` -> `collection_funcs` due to it used for both `Array` and `Map`

On the other hand, this pr corrects the description of `ExpressionDescription#group`, `collection_funcs` is divided into six sub-groups now:

- array_funcs
- csv_funcs
- generator_funcs
- json_funcs
- lambda_funcs
- map_funcs

### Why are the changes needed?
Correct group of collection_funcs

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass Github Actions

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala [hunk 1]
```diff
@@ -158,7 +158,7 @@ object Size {
        4
   """,
   since = "3.3.0",
-  group = "collection_funcs")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:161]: We should actually replace all of `collection_funcs` case: https://github.com/apache/spark/blob/af1dba7ca501fd9372b158793119163e3fcd1f24/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionDescription.java#L110-L116

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:161]: For others, maybe use `agg_funcs` (?).

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:161]: Should probably pick something closer for others for `Size` or add a new group if they do not fit.

Comment (LuciferYang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:161]: https://github.com/apache/spark/pull/30867 

image


From SPARK-33124, I think `Size` should use `collection_funcs`, it is similar to `Reverse` and `ElementAt`

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:161]: Hm, I see. Okay, makes sense to me

**Comment by LuciferYang** [comment]
I will close this pr today if it is not worth changing

**Pull request closed without merging**